boot/duneboot.ml

```diff
@@ -664,7 +664,7 @@ let scheduler parallel_count name exe external_includes external_libraries
           let task =
             let cwd = Sys.getcwd () in
             Sys.chdir build_dir;
-            Printf.printf "%s -c -g -no-alias-deps -w -49 -I +threads %s\n"
+            Printf.printf "%s -c -g -no-alias-deps -w -49 -I +threads %s\n%!"
```

Comment:
I should have put a COMBAK on this - the flush considerably slows the build down (same with `dune` itself)

Reply:
(I've also left things hard-coded in the string, but that's another story...)

Reply:
Ok. I suppose we can do the same as for Dune then: we only print command that fail

Reply:
Or we can also print nothing. The output of the compiler already contains `File ...`

Reply:
On my Ubuntu 19.04 4-core VirtualBox (so at the "console"): 

```
dra@disco:~/dune-vanilla$ time make
…
real	0m36.229s
user	1m5.748s
sys	0m23.141s

dra@disco:~/dune-vanilla$ git clean -dfx &>/dev/null
dra@disco:~/dune-vanilla$ time make &>/dev/null

real	0m21.695s
user	0m42.232s
sys	0m11.225s
```

On our EPYCs (via SSH):

```
dra27@bremusa:~/dune$ time make
…
real    0m10.891s
user    0m34.895s
sys     0m15.863s

dra27@bremusa:~/dune$ git clean -dfx &>/dev/null
dra27@bremusa:~/dune$ time make &>/dev/null

real    0m10.984s
user    0m34.487s
sys     0m16.254s
```

but on its console, without piping to `/dev/null`:

```
real    0m14.198s
user    0m35.433s
sys     0m16.273s
```

I think that when `stdout` is a tty, `dune` should be more conservative about flushing.

Reply:
I just hid the compilation commands. We could just dump them into a log file.